public/app/features/logs/utils.ts

```diff
@@ -45,7 +45,7 @@ export function getLogLevel(line: string): LogLevel {
 }
 
 export function getLogLevelFromKey(key: string | number): LogLevel {
-  const level = (LogLevel as any)[key.toString().toLowerCase()];
+  const level = LogLevel[key.toString().toLowerCase() as keyof typeof LogLevel];
```

Comment:
Does `keyof typeof LogLevel` really represent a key from the enum? I can't get my head around the `keyof typeof` part, since `typeof` is object?

Reply:
https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-compile-time

Reply:
TIL. Thanks!

Reply:
I love that the example in https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-compile-time is almost exactly the same code as we have - LogLevel. 😆 Thank you for sharing @tskarhed .

Reply:
Maybe they took inspiration from here 😂